src/app/recipes/resolvers/recipe.resolver.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { Resolve } from '@angular/router';
+import { Store } from '@ngrx/store';
+import RecipesActions from '@store/recipes/recipes.actions';
+import RecipesSelectors from '@store/recipes/recipes.selector';
+import { AppState } from '@store/store';
+import { finalize, first, Observable, tap } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class RecipeResolver implements Resolve<any> {
+	loading = false;
+
+	constructor(private store: Store<AppState>) {}
+
+	resolve(): Observable<any> {
+		return this.store.select(RecipesSelectors.selectLoadingIsComplied).pipe(
```

Comment:
what will happen if there's an error? You can add in future a redirect to error page or something here

Reply:
I only are dispatch recipeDownloadFaild action. But in the futer I can add effect and handle it some way.